oracle/src/currency.rs

```diff
@@ -46,9 +46,12 @@ impl<Currency> From<(Currency, Currency)> for CurrencyPair<Currency> {
     }
 }
 
-impl<Currency: PartialEq> CurrencyPair<Currency> {
+impl<Currency: PartialEq + ToString> CurrencyPair<Currency> {
     pub fn contains(&self, currency: &Currency) -> bool {
-        &self.base == currency || &self.quote == currency
+        let strip = |x: &Currency| x.to_string().split("=").next().unwrap().to_string();
+        let stripped_currency = strip(currency);
```

Comment:
I think we can do this during deserialization, but I will merge this PR and experiment.